diff --git a/src/plugins/platforms/cocoa/qcocoacolordialoghelper.mm b/src/plugins/platforms/cocoa/qcocoacolordialoghelper.mm
index 47f9539..1d588b4 100644
--- a/src/plugins/platforms/cocoa/qcocoacolordialoghelper.mm
+++ b/src/plugins/platforms/cocoa/qcocoacolordialoghelper.mm
@@ -129,10 +129,18 @@ QT_NAMESPACE_ALIAS_OBJC_CLASS(QNSColorPanelDelegate);
     [super dealloc];
 }
 
-- (void)setDialogHelper:(QCocoaColorDialogHelper *)helper
+- (void)setDialogHelper:(QCocoaColorDialogHelper *)helper andSetShowsAlpha:(BOOL)setShowsAlpha
 {
     mHelper = helper;
-    [mColorPanel setShowsAlpha:mHelper->options()->testOption(QColorDialogOptions::ShowAlphaChannel)];
+
+    // Workaround for Apple rdar://25792119: If you invoke
+    // -setShowsAlpha: multiple times before showing the color
+    // picker, its height grows irrevocably.  Instead, only
+    // invoke it once we know its final status.
+    if (setShowsAlpha)
+        [mColorPanel setShowsAlpha:mHelper->options()->testOption(
+                QColorDialogOptions::ShowAlphaChannel)];
+
     if (mHelper->options()->testOption(QColorDialogOptions::NoButtons)) {
         [self restoreOriginalContentView];
     } else if (!mStolenContentView) {
@@ -392,9 +400,9 @@ public:
         [mDelegate release];
     }
 
-    void init(QCocoaColorDialogHelper *helper)
+    void init(QCocoaColorDialogHelper *helper, bool setShowsAlpha)
     {
-        [mDelegate setDialogHelper:helper];
+        [mDelegate setDialogHelper:helper andSetShowsAlpha:setShowsAlpha];
     }
 
     void cleanup(QCocoaColorDialogHelper *helper)
@@ -483,7 +491,7 @@ bool QCocoaColorDialogHelper::show(Qt::WindowFlags, Qt::WindowModality windowMod
 {
     if (windowModality == Qt::WindowModal)
         windowModality = Qt::ApplicationModal;
-    sharedColorPanel()->init(this);
+    sharedColorPanel()->init(this, true);
     return sharedColorPanel()->show(windowModality, parent);
 }
 
@@ -494,7 +502,7 @@ void QCocoaColorDialogHelper::hide()
 
 void QCocoaColorDialogHelper::setCurrentColor(const QColor &color)
 {
-    sharedColorPanel()->init(this);
+    sharedColorPanel()->init(this, false);
     sharedColorPanel()->setCurrentColor(color);
 }
 
