--- src/plugins/platforms/cocoa/qcocoafiledialoghelper.mm.orig	2020-07-31 13:44:12.000000000 -0400
+++ src/plugins/platforms/cocoa/qcocoafiledialoghelper.mm	2020-07-31 14:14:43.000000000 -0400
@@ -152,7 +152,7 @@ QT_NAMESPACE_ALIAS_OBJC_CLASS(QNSOpenSav
         mCurrentSelection = new QString(sel.absoluteFilePath());
     }
 
-    [mSavePanel setTitle:options->windowTitle().toNSString()];
+    [mSavePanel setMessage:options->windowTitle().toNSString()];
     [self createPopUpButton:selectedVisualNameFilter hideDetails:options->testOption(QFileDialogOptions::HideNameFilterDetails)];
     [self createTextField];
     [self createAccessory];
@@ -443,8 +443,8 @@ static QString strippedText(QString s)
     [mSavePanel setCanCreateDirectories:!(mOptions->testOption(QFileDialogOptions::ReadOnly))];
     [mOpenPanel setAllowsMultipleSelection:(fileMode == QFileDialogOptions::ExistingFiles)];
     [mOpenPanel setResolvesAliases:!(mOptions->testOption(QFileDialogOptions::DontResolveSymlinks))];
-    [mOpenPanel setTitle:mOptions->windowTitle().toNSString()];
-    [mSavePanel setTitle:mOptions->windowTitle().toNSString()];
+    [mOpenPanel setMessage:mOptions->windowTitle().toNSString()];
+    [mSavePanel setMessage:mOptions->windowTitle().toNSString()];
     [mPopUpButton setHidden:chooseDirsOnly];    // TODO hide the whole sunken pane instead?
 
     if (mOptions->acceptMode() == QFileDialogOptions::AcceptSave) {
@@ -655,7 +655,7 @@ void QCocoaFileDialogHelper::setFilter()
     if (!mDelegate)
         return;
     const SharedPointerFileDialogOptions &opts = options();
-    [mDelegate->mSavePanel setTitle:opts->windowTitle().toNSString()];
+    [mDelegate->mSavePanel setMessage:opts->windowTitle().toNSString()];
     if (opts->isLabelExplicitlySet(QFileDialogOptions::Accept))
         [mDelegate->mSavePanel setPrompt:[mDelegate strip:opts->labelText(QFileDialogOptions::Accept)]];
     if (opts->isLabelExplicitlySet(QFileDialogOptions::FileName))
