--- src/network/ssl/qsslconfiguration.h.orig	2020-08-01 17:09:00.000000000 -0400
+++ src/network/ssl/qsslconfiguration.h	2020-08-01 17:09:13.000000000 -0400
@@ -152,7 +152,7 @@ public:
 
     QMap<QByteArray, QVariant> backendConfiguration() const;
     void setBackendConfigurationOption(const QByteArray &name, const QVariant &value);
-    void setBackendConfiguration(const QMap<QByteArray, QVariant> &backendConfiguration = QMap<QByteArray, QVariant>());
+    void setBackendConfiguration(const QMap<QByteArray, QVariant> &backendConfiguration);
 
     static QSslConfiguration defaultConfiguration();
     static void setDefaultConfiguration(const QSslConfiguration &configuration);
